radix-engine/src/blueprints/consensus_manager/consensus_manager.rs

```diff
@@ -46,8 +46,44 @@ pub struct Validator {
 
 #[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
 pub struct CurrentValidatorSetSubstate {
-    /// The current validator set in the epoch, ordered by stake descending.
-    pub validator_set: IndexMap<ComponentAddress, Validator>,
+    pub validator_set: ActiveValidatorSet,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
+#[sbor(transparent)]
+pub struct ActiveValidatorSet {
+    /// The validators in the set, ordered by stake descending.
+    pub validators_by_stake_desc: IndexMap<ComponentAddress, Validator>,
+}
+
+impl ActiveValidatorSet {
+    pub fn get_by_index(&self, index: ValidatorIndex) -> Option<(&ComponentAddress, &Validator)> {
+        self.validators_by_stake_desc.get_index(index as usize)
+    }
+
+    pub fn get_by_address(&self, address: &ComponentAddress) -> Option<&Validator> {
+        self.validators_by_stake_desc.get(address)
+    }
+
+    pub fn get_by_public_key(
```

Comment:
(minor) I imagine we want to use it only for tests (= worth annotating about poor runtime performance)